massa-bootstrap/src/listener.rs

```diff
@@ -85,11 +87,25 @@ impl BSEventPoller for BootstrapTcpListener {
             return Ok(PollEvent::Stop);
         }
 
-        // Ther could be more than one connection ready, but we want to re-check for the stop
-        // signal after processing each connection.
-        Ok(PollEvent::NewConnection(
-            self.server.accept().map_err(BootstrapError::from)?,
-        ))
+        let mut results = Vec::with_capacity(self.events.iter().count());
+
+        // Process each event.
+        for event in self.events.iter() {
+            match event.token() {
+                NEW_CONNECTION => {
+                    results.push(self.server.accept()?);
+                }
+                _ => unreachable!(),
+            }
+        }
+
+        // We need to have an accept() error with WouldBlock, otherwise polling may not raise any new events.
+        // See https://users.rust-lang.org/t/why-mio-poll-only-receives-the-very-first-event/87501
```

Comment:
Consider providing context about why we cannot add it to the results vec, and what changes might be implemented to improve things here.